exercises/strings/rotN/rotN.js

```diff
@@ -15,15 +15,37 @@
  * @returns {string} A lowercase copy of the input string
  */
 
-function rotN(string) {
+function rotN(string, n) {
   // This is your job. :)
+  var i;
+  var new_string = '';  
+  for (i=0; i<string.length; i++){
+    var char = string.charCodeAt(i);
+    // lower case letter
+    if ((char >= 97) && (char <= 122)){
+      new_string += String.fromCharCode(((char - 96 + n) % 26) + 96);
+    }
+    // upper case
+    else if ((char >= 65) && (char <= 90)){
+      new_string += String.fromCharCode(((char - 64 + n) % 26) + 64);
+    }
+    // any other character
+    else{
+      new_string += string.charAt(i);
+    }
+  }
+  return new_string;
+
 }
 
 if (require.main === module) {
   console.log('Running sanity checks for rotN:');
 
   // Add your own sanity checks here.
   // How else will you be sure your code does what you think it does?
+  console.log(rotN('Hello, world!', 9)); // => 'Qnuux, fxaum!'
```

Comment:
Consider putting the comparison directly in the `console.log` call so that you see `true` if the return value is what you expect and `false` otherwise. Without that, it's basically impossible to tell whether you're reading the output correctly.